src/uvc-sensor.cpp

```diff
@@ -133,22 +135,31 @@ void uvc_sensor::open( const stream_profiles & requests )
                         return;
                     }
 
-                    const auto && fr = generate_frame_from_data( f,
+                    auto && fr = generate_frame_from_data( f,
                                                                  system_time,
                                                                  _timestamp_reader.get(),
                                                                  last_timestamp,
                                                                  last_frame_number,
                                                                  req_profile_base );
-                    const auto && timestamp_domain = _timestamp_reader->get_frame_timestamp_domain( fr );
+                    auto timestamp_domain = _timestamp_reader->get_frame_timestamp_domain( fr );
                     auto bpp = get_image_bpp( req_profile_base->get_format() );
-                    auto && frame_counter = fr->additional_data.frame_number;
-                    auto && timestamp = fr->additional_data.timestamp;
+                    auto & frame_counter = fr->additional_data.frame_number;
+                    auto & timestamp = fr->additional_data.timestamp;
 
                     // D457 development
                     size_t expected_size;
                     auto && msp = As< motion_stream_profile, stream_profile_interface >( req_profile );
                     if( msp )
+                    {
                         expected_size = 64;  // 32; // D457 - WORKAROUND - SHOULD BE REMOVED AFTER CORRECTION IN DRIVER
+                        auto pixels = (uint8_t *)f.pixels;
```

Comment:
Please add comment explaining why we override the original counter

Reply:
Added